subprojects/docs/src/docs/userguide/jvm/java_testing.adoc

```diff
@@ -618,7 +618,11 @@ In this case, Gradle will mark the skipped tests as "SKIPPED" rather than exclud
 
 In well-defined builds, you can rely on Gradle to only run tests if the tests themselves or the production code change. However, you may encounter situations where the tests rely on a third-party service or something else that might change but can't be modeled in the build.
 
-You can force tests to run in this situation by cleaning the output of the relevant `Test` task — say `test` — and running the tests again, like so:
+You can always use the `--rerun` <<command_line_interface.adoc#sec:builtin_task_options,built-in task option>> to force a task to rerun.
+
+    gradle test --rerun
+
+Alternatively, if <<build_cache.adoc#sec:build_cache_enable,build caching>> is not enabled, you can also force tests to run by cleaning the output of the relevant `Test` task — say `test` — and running the tests again, like so:
```

Comment:
:question: I wonder if we should not remove the alternative way now. `--rerun` feels like a better replacement. Recommending a single way to rerun tests sounds better to me. WDYT?

Reply:
I guess I have a tendency to lean on the conservative side (well, at least product-evolution wise).  Since they are not exactly equivalent alternatives, I wondered if there could be scenarios that would be better served by the `clean<Task>` mechanism, and if so, thought it would be a good discovery opportunity for those features. I could be convinced otherwise, if you feel strongly about it, I am on the fence here, leaning towards taking minimal steps.

Reply:
@eskatos I am going ahead and merge this (minus the wrapper version change, which Stefan suggested but I feel may cause unexpected consequences, and I could always do that as a separate PR). If you feel strongly about this, I will be happy to create another change set with that change alone, just let me know!

Reply:
Sure, this can be merged as is.

Reply:
I think we have too many places where there are several ways to do the same thing. Clean task rules are already documented in the Java Plugin chapter https://docs.gradle.org/current/userguide/java_plugin.html#sec:java_tasks
I would vote for removing it from here and recommend a single way to rerun the Test task.